eq-author/src/components/buttons/ExternalLinkButton/index.js

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+import { radius, colors, focusStyle } from "constants/theme";
+
+import { ReactComponent as Icon } from "App/QuestionnaireDesignPage/MainNavigation/icons/view-survey-icon.svg";
+
+const propTypes = {
+  text: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired,
+};
+
+const Button = styled.button`
+  display: inline-flex;
+  flex: 0 0 auto;
+  color: ${colors.primary};
+  background-color: transparent;
+  padding: 0.75em 1em;
+  border-radius: ${radius};
+  font-size: 1em;
+  font-weight: 600;
+  cursor: pointer;
+  line-height: 1.1;
+  justify-content: center;
+  -webkit-font-smoothing: antialiased;
+  text-rendering: optimizeLegibility;
+  text-decoration: none;
+  transition: all 100ms ease-out;
+  letter-spacing: 0;
+  position: relative;
+  overflow: hidden;
+  border: 1px solid ${colors.primary};
+  margin-top: 8px;
+
+  &[disabled] {
+    pointer-events: none;
+    opacity: 0.6;
+  }
+
+  &:hover {
+    color: ${colors.white};
+    background-color: ${colors.primary};
+  }
+
+  &:focus-within {
+    ${focusStyle}
+  }
+`;
+
+const ExternalIcon = styled(Icon)`
+  margin-left: 12px;
+  path {
+    fill: currentColor;
+  }
+`;
+
+const EnabledLinkButton = Button.withComponent("a");
+const DisabledLinkButton = Button.withComponent("div");
+
+class LinkButtonWithIcon extends React.Component {
+  static propTypes = {
+    ...propTypes,
+    text: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+  };
+
+  render() {
+    const { text, url, disabled, ...otherProps } = this.props;
+    return disabled ? (
+      <DisabledLinkButton aria-disabled="true" disabled {...otherProps}>
+        <span>{text}</span>
+        <ExternalIcon />
+      </DisabledLinkButton>
+    ) : (
+      <EnabledLinkButton
+        {...otherProps}
+        href={url}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        <span>{text}</span>
+        <ExternalIcon />
+      </EnabledLinkButton>
+    );
+  }
+}
```

Comment:
This component can be converted into a functional component, as we write newly developed components in Author as functional components - we have been looking at reducing the number of class components in Author, due to being more complex to maintain.

Functional components allow us to make use of React hooks, which can reduce the number of lines of code required, improving readability and also maintainability. Functional components will also simplify the process of updating Author's Node version.

Reply:
Converted to function with Sam's help